BSP/Src/BSP_I2C.c

```diff
@@ -0,0 +1,180 @@
+
+#include "BSP_I2C.h"
+#include "BSP.h"
+#include "FreeRTOS.h"
+#include "queue.h"
+
+#define I2C_QUEUE_SIZE   64
+#define I2C_ITEM_LENGTH   sizeof(uint8_t)
+
+static StaticQueue_t __DataStaticQueue;
+static StaticQueue_t __MetaStaticQueue;
+static StaticQueue_t __ReceiveStaticQueue;
+
+
+uint8_t __DataBufferStorage[I2C_QUEUE_SIZE * I2C_ITEM_LENGTH];
+uint8_t __MetaBufferStorage[I2C_QUEUE_SIZE * I2C_ITEM_LENGTH * 2];
+uint8_t __ReceiveBufferStorage[I2C_QUEUE_SIZE * I2C_ITEM_LENGTH];
+uint8_t __ReceiveRawData[I2C_QUEUE_SIZE * I2C_ITEM_LENGTH];
+
+static QueueHandle_t I2C_DataQueue;
+//addresses, registers, and lengths
+static QueueHandle_t I2C_MetaQueue;
+
+static QueueHandle_t I2C_ReceiveQueue; 
+
+BSP_Status BSP_I2C_Init() {
+    //QUEUES
+    I2C_DataQueue = xQueueCreateStatic(I2C_QUEUE_SIZE, I2C_ITEM_LENGTH, __DataBufferStorage, &__DataStaticQueue);
+    I2C_MetaQueue = xQueueCreateStatic(I2C_QUEUE_SIZE * 2, I2C_ITEM_LENGTH, __MetaBufferStorage, &__MetaStaticQueue);
+    I2C_ReceiveQueue = xQueueCreateStatic(I2C_QUEUE_SIZE, I2C_ITEM_LENGTH, __ReceiveBufferStorage, &__ReceiveStaticQueue);
+
+
+    GPIO_InitTypeDef gpio_struct;
+    I2C_HandleTypeDef i2c_struct;
+
+    if (__HAL_RCC_I2C3_IS_CLK_DISABLED()) {
+        __HAL_RCC_I2C3_CLK_ENABLE();
+    }
+    if (__HAL_RCC_GPIOA_IS_CLK_DISABLED()) {
+        __HAL_RCC_GPIOA_CLK_ENABLE();
+    }
+    if (__HAL_RCC_GPIOC_IS_CLK_DISABLED()) {
+        __HAL_RCC_GPIOC_CLK_ENABLE();
+    }
+
+    // i2c config
+    i2c_struct.Init.ClockSpeed = I2C_CLOCK_SPEED;
```

Comment:
need to verify the init function, initialization is in a weird state between not doing the GPIO and only initializing the peripheral due to the standard config for this repo that's being reviewed